corehq/apps/reports/filters/users.py

```diff
@@ -328,7 +330,24 @@ def user_es_query(cls, domain, mobile_user_and_group_slugs):
             user_type_filters.append(user_es.demo_users())
 
         q = user_es.UserES().domain(domain)
-        if HQUserType.REGISTERED in user_types:
+        if not can_access_all_locations:
+            # Make sure the passed in locations are accessible
+            location_ids = (SQLLocation.active_objects
+                            .get_locations(location_ids)
+                            .accessible_to_user(domain, request_user)
+                            .location_ids())
+            if HQUserType.REGISTERED in user_types:
+                all_accessible_loc_ids = (SQLLocation.active_objects
+                                          .accessible_to_user(domain, request_user)
+                                          .location_ids())
+                q = q.location(all_accessible_loc_ids)
+                return q.OR(
```

Comment:
what's the meaning of this OR - does this mean it ORs the args passed to it and ANDs the result to the main query?

Reply:
That's correct